### ISSUE TITLE
Failed Installation OSX 10.14.6

### ISSUE THREAD

**Issue by akhst7** [jlmelville/uwot#28]
Hi, 

Installation of uwot,  by` install_github()` or `install()` failed with the same error. Even `R --vanilla` failed to install it with the same error;

```
clang++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -L/usr/local/Cellar/r/3.6.1_1/lib/R/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -o uwot.so RcppExports.o connected_components.o gradient.o nn_parallel.o optimize.o perplexity.o sampler.o smooth_knn.o supervised.o transform.o All my packages loaded Tue Aug 27 23:05:12 2019 -L/usr/local/Cellar/r/3.6.1_1/lib/R/lib -lR -lintl -Wl,-framework -Wl,CoreFoundation
clang-8: error: no such file or directory: 'All'
clang-8: error: no such file or directory: 'my'
clang-8: error: no such file or directory: 'packages'
clang-8: error: no such file or directory: 'loaded'
clang-8: error: no such file or directory: 'Tue'
clang-8: error: no such file or directory: 'Aug'
clang-8: error: no such file or directory: '27'
clang-8: error: no such file or directory: '23:05:14'
clang-8: error: no such file or directory: '2019'
make: *** [uwot.so] Error 1
```
I am using OSX Mojave, 

```
sw_vers
ProductName:	Mac OS X
ProductVersion:	10.14.6
BuildVersion:	18G87
```
gcc version (not apples);

gcc -v
```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/local/Cellar/gcc/9.2.0/libexec/gcc/x86_64-apple-darwin18/9.2.0/lto-wrapper
Target: x86_64-apple-darwin18
Configured with: ../configure --build=x86_64-apple-darwin18 --prefix=/usr/local/Cellar/gcc/9.2.0 --libdir=/usr/local/Cellar/gcc/9.2.0/lib/gcc/9 --disable-nls --enable-checking=release --enable-languages=c,c++,objc,obj-c++,fortran --program-suffix=-9 --with-gmp=/usr/local/opt/gmp --with-mpfr=/usr/local/opt/mpfr --with-mpc=/usr/local/opt/libmpc --with-isl=/usr/local/opt/isl --with-system-zlib --with-pkgversion='Homebrew GCC 9.2.0' --with-bugurl=https://github.com/Homebrew/homebrew-core/issues --disable-multilib --with-native-system-header-dir=/usr/include --with-sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
Thread model: posix
gcc version 9.2.0 (Homebrew GCC 9.2.0)
```
```
clang -v
clang version 8.0.1 (tags/RELEASE_801/final)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /usr/local/Cellar/llvm/8.0.1/bin
```
Any help or pointer to fix this is really appreciated.

**Comment by SamGG** [comment]
Don't have a Mac either, but did you notice that in your command line there is in fact the string `All my packages loaded Tue Aug 27 23:05:12 2019` which has no meaning for the compiler? I don't know where it came from, but it's funny.

**Comment by akhst7** [comment]
It did compile in another Mac with the similar set-up but not sure what is causing this.  I have seen this before with R loading some sort of package info during completion.   

Anyway could you tell me the way to  manually compile and generate binary of uwot without using R   ? 

Thanks.

**Comment by LTLA** [comment]
I'm using a Mac, and it works fine for me.

```
$ clang --version
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.6
BuildVersion:	17G8030
```

As @SamGG, the error message is very unusual. Can you clone the repo and post the full output of `R CMD INSTALL --preclean path/to/uwot/directory`?

**Comment by akhst7** [comment]
@LTLA 

So  I did R `CMD check` and here is an output;

```
%more 00check.log
* using log directory ‘/Volumes/Documents/Users/akihoji/uwot-master.Rcheck’
* using R version 3.6.1 (2019-07-05)
* using platform: x86_64-apple-darwin18.6.0 (64-bit)
* using session charset: UTF-8
* checking for file ‘uwot-master/DESCRIPTION’ ... ERROR
Required fields missing or empty:
  ‘Author’ ‘Maintainer’
* DONE
Status: 1 ERROR
```
OK then I tried R CMD INSTALL -fake  and 

```
% R CMD INSTALL --fake  Downloads/uwot-master
Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install


All my packages loaded Thu Aug 29 20:00:12 2019* installing to library ‘/Volumes/Documents/Users/akihoji/Library/R/3.x/library’
* installing *source* package ‘uwot’ ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install


All my packages loaded Thu Aug 29 20:00:14 2019
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install


All my packages loaded Thu Aug 29 20:00:18 2019
** testing if installed package can be loaded from final location
Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install


All my packages loaded Thu Aug 29 20:00:20 2019
** testing if installed package keeps a record of temporary installation path
* DONE (uwot)
```
This installed uwot but missing  full functionalities as expected. 

R CMD INSTALL failed and the output is ;

```
% R CMD INSTALL --preclean Downloads/uwot-master
Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install


All my packages loaded Thu Aug 29 20:41:43 2019* installing to library ‘/Volumes/Documents/Users/akihoji/Library/R/3.x/library’
* installing *source* package ‘uwot’ ...
** using staged installation
** libs
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c RcppExports.cpp -o RcppExports.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c connected_components.cpp -o connected_components.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c gradient.cpp -o gradient.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c nn_parallel.cpp -o nn_parallel.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c optimize.cpp -o optimize.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c perplexity.cpp -o perplexity.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c sampler.cpp -o sampler.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c smooth_knn.cpp -o smooth_knn.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c supervised.cpp -o supervised.o
clang++ -std=gnu++11 -I"/usr/local/Cellar/r/3.6.1_1/lib/R/include" -DNDEBUG  -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/Rcpp/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppProgress/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppParallel/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/RcppAnnoy/include" -I"/Volumes/Documents/Users/akihoji/Library/R/3.x/library/dqrng/include" -I/usr/local/opt/gettext/include -I/usr/local/opt/readline/include -I/usr/local/include -DRCPP_PARALLEL_USE_TBB=1 -DSTRICT_R_HEADERS -fPIC  -g -O2  -c transform.cpp -o transform.o
Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install

Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install

clang++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -L/usr/local/Cellar/r/3.6.1_1/lib/R/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/readline/lib -L/usr/local/lib -o uwot.so RcppExports.o connected_components.o gradient.o nn_parallel.o optimize.o perplexity.o sampler.o smooth_knn.o supervised.o transform.o All my packages loaded Thu Aug 29 20:42:39 2019 -L/usr/local/Cellar/r/3.6.1_1/lib/R/lib -lR -lintl -Wl,-framework -Wl,CoreFoundation
clang-8: error: no such file or directory: 'All'
clang-8: error: no such file or directory: 'my'
clang-8: error: no such file or directory: 'packages'
clang-8: error: no such file or directory: 'loaded'
clang-8: error: no such file or directory: 'Thu'
clang-8: error: no such file or directory: 'Aug'
clang-8: error: no such file or directory: '29'
clang-8: error: no such file or directory: '20:42:41'
clang-8: error: no such file or directory: '2019'
make: *** [uwot.so] Error 1
ERROR: compilation failed for package ‘uwot’
* removing ‘/Volumes/Documents/Users/akihoji/Library/R/3.x/library/uwot’
```

**Comment by LTLA** [comment]
Probably should have mentioned to do `R CMD build`, and then `R CMD INSTALL` on the tarball.

But that's probably not the real problem. Your R installation seems to be doing strange things. Why is it giving out these messages:

```
Loading required package: usethis

Attaching package: ‘BiocManager’

The following object is masked from ‘package:devtools’:

    install


All my packages loaded ...
```

You probably have an `.Rprofile` file somewhere that is triggering these messages - you might consider getting rid of it. I don't understand how it would interfere with clang, but clearly it is doing so, so the first task is to simplify the problem.

**Comment by akhst7** [comment]
I did look for .Rprofile eveywhere but could not find one that was causing the problem.    As you mentioned, devtools' call for usethis pops up and appears to interfere with clang which I am not sure how.

So, I have two R libraries, system and my custom.  My custom library has everything that I have installed including devtools and BiocManager and I though autoloading devtools and BIocManger invoked the call for "usethis."  So I tried to install uwot in the system library  and uwot installation went without a hitch.  

I've installed a bunch of other packages (including singleR) from github without a hitch.  There must be something in the code that is provoking the loading of "usethis".  

Anyways, thanks for all the help.
